src/dao/UserLink.ts

```diff
@@ -0,0 +1,23 @@
+import {Redis} from "../database/Redis";
+import {genDbKey} from "../utility/Utility";
+
+export const KEY_USER_FOLLOWER_ID = "User:Follower:%ID%";
+export const KEY_USER_FOLLOWING_ID = "User:Following:%ID%";
+
+export const followUser = async (userId: string, targetId: string) => {
+  await Redis.get().sadd(genDbKey(KEY_USER_FOLLOWER_ID, {"%ID%": targetId}), userId);
```

Comment:
what if the 1st `sadd` succeeds but the 2nd one fails somehow, how to mitigate the data inconsistency in some edge-cases?

Reply:
One solution here is: we could use `transaction` of redis, like: 

``` typescript
await Redis.get().multi().sadd().sadd().exec();
```

And the transaction of redis is `NOT SAFE` for atomic operation. See [the official doc](https://redis.io/topics/transactions#errors-inside-a-transaction).

> It's important to note that even when a command fails, all the other commands in the queue are processed – Redis will not stop the processing of commands.

---

The other solution here is: lua script:

Since there is no rollback of official redis commands. If we need to keep these two commands all success or all rolled back when any error encountered, we could use self defined lua script to do it. See [official doc](https://redis.io/commands/eval).